source/tutorials/file-sets.md

```diff
@@ -307,126 +375,168 @@ error: lib.fileset.difference: Second argument (negative set)
-        ./default.nix
-        ./package.nix
-        ./nix
-      ]);
+Create a file set containing a union of the files to exclude (`fs.unions [ ... ]`), and subtract it (`difference`) from the complete directory (`./.`):
+
+```{code-block} diff
+:caption: build.nix
+ { stdenv, lib }:
+ let
+   fs = lib.fileset;
+-  sourceFiles = fs.difference ./. (fs.maybeMissing ./result);
++  sourceFiles =
++    fs.difference
++      ./.
++      (fs.unions [
++        (fs.maybeMissing ./result)
++        ./default.nix
++        ./build.nix
++        ./nix
++      ]);
+ in
 ```
 
-This also gives us the opportunity to show the [`fileFilter`](https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.fileset.fileFilter) function,
-which as the name implies, allows filtering the files in a local path.
-We use it to select all files whose name ends with `.nix`:
+Changing any of the excluded files now doesn't necessarily require a rebuild anymore.
 
-```{code-block} nix
-:caption: package.nix
-  sourceFiles =
-    fs.difference
-      ./.
-      (fs.unions [
-        (fs.maybeMissing ./result)
-        (fs.fileFilter (file: lib.hasSuffix ".nix" file.name) ./.)
-        ./nix
-      ]);
-```
+Check it and modify one of the excluded files again:
 
-Changing any of the removed files now doesn't necessarily require a rebuild anymore:
+```
+$ echo >> build.nix
+```
 
-```shell-session
+```
 $ nix-build
-trace: /home/user/select
-trace: - string.txt (regular)
-/nix/store/clrd19vn5cv6n7x7hzajq1fv43qig7cp-filesets
+trace: /home/user/fileset
+trace: - hello.txt (regular)
+trace: - world.txt (regular)
+/nix/store/ckn40y7hgqphhbhyrq64h9r6rvdh973r-fileset
+```
+
+## Filter
+
+Dealing with a large number of files, independent of their location, can be done programmatically with the [`fileFilter`](https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.fileset.fileFilter) function.
 
-$ echo "# Just a comment" >> package.nix
+Use it to select all files with a name ending in `.nix`:
 
+```{code-block} diff
+:caption: build.nix
+ { stdenv, lib }:
+ let
+   fs = lib.fileset;
+   sourceFiles =
+     fs.difference
+       ./.
+       (fs.unions [
+         (fs.maybeMissing ./result)
+-        ./default.nix
+-        ./build.nix
++        (fs.fileFilter (file: lib.hasSuffix ".nix" file.name) ./.)
+         ./nix
+       ]);
+ in
+```
+
+This does not change the result:
+
+```shell-session
 $ nix-build
-trace: /home/user/select
-trace: - string.txt (regular)
-/nix/store/clrd19vn5cv6n7x7hzajq1fv43qig7cp-filesets
+trace: /home/user/fileset
+trace: - hello.txt (regular)
+trace: - world.txt (regular)
+/nix/store/ckn40y7hgqphhbhyrq64h9r6rvdh973r-fileset
 ```
 
-Notable with this approach is that new files added to the current directory are _included_ by default.
+Notably, with this approach, new files added to the source directory are _included_ by default.
 Depending on your project, this might be a better fit than the alternative in the next section.
```

Comment:
```suggestion
Notably, with this approach of using `difference ./.`, we're explicitly selecting the files to _exclude_, which means that new files added to the source directory are _included_ by default.
Depending on your project, this might be a better fit than the alternative in the next section.
```

Reply:
```suggestion
Notably, the approach of using `difference ./.` explicitly selects the files to _exclude_, which means that new files added to the source directory are included by default.
Depending on your project, this might be a better fit than the alternative in the next section.
```

Removing the ominous "we" and the emphasis on "including", which was a bit irritating already because the section is "Union (exclude)" - now that's resolved, yay!